webui/react/src/components/RunActionDropdown.tsx

```diff
@@ -0,0 +1,190 @@
+import { GridCell } from '@glideapps/glide-data-grid';
+import { ContextMenuCompleteHandlerProps } from 'hew/DataGrid/contextMenu';
+import Dropdown, { DropdownEvent, MenuItem } from 'hew/Dropdown';
+import { useModal } from 'hew/Modal';
+import { useToast } from 'hew/Toast';
+import useConfirm from 'hew/useConfirm';
+import { copyToClipboard } from 'hew/utils/functions';
+import React, { MouseEvent, useCallback, useMemo } from 'react';
+
+import usePermissions from 'hooks/usePermissions';
+import FlatRunMoveModalComponent from 'pages/FlatRuns/FlatRunMoveModal';
+import { handlePath } from 'routes/utils';
+import { archiveRuns, deleteRuns, killRuns, unarchiveRuns } from 'services/api';
+import { FlatRun, FlatRunAction, ValueOf } from 'types';
+import handleError, { ErrorLevel, ErrorType } from 'utils/error';
+import { getActionsForFlatRun } from 'utils/flatRun';
+import { capitalize } from 'utils/string';
+
+import { FilterFormSetWithoutId } from './FilterForm/components/type';
+
+interface Props {
+  children?: React.ReactNode;
+  cell?: GridCell;
+  run: FlatRun;
+  link?: string;
+  makeOpen?: boolean;
+  onComplete?: ContextMenuCompleteHandlerProps<FlatRunAction, void>;
+  onLink?: () => void;
+  onVisibleChange?: (visible: boolean) => void;
+  projectId: number;
+  filterFormSetWithoutId: FilterFormSetWithoutId;
+}
+
+const Action = {
+  Copy: 'Copy Value',
```

Comment:
bug?: when i copy value of `User`, `userId is copied`. ideally username (displayname?) should be copied. or avoid copying user value?